### PR TITLE
feat: add Layout component

### PR DESCRIPTION
**Pull request by jmfrancois**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

Layout is done in the App so the App handle html markup + css setup

**What is the new behavior?**

You can use layout to create a new page.
This PR provide two layouts, but more can be added.

```javascript
<Layout
	header={header}
	mode="OneColumn"
	one={contentOne}
/>
<Layout
	header={header}
	mode="TwoColumns"
	one={contentOne}
	two={contentTwo}
/>
```

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

screen shot 2016-10-24 at 15 57 19

This kind of components will be used as 'view' in a react-cmf app

### CODE REVIEW

File: src/Layout/Layout.scss [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Layout styles
+.app {
+	height: 100%;
+	display: flex;
+	flex-direction: column;
+}
+
+.header {
+	flex: 0 0 49px;
```
Comment (frassinier) [inline src/Layout/Layout.scss]: 49?

File: src/Layout/Layout.scss [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// Layout styles
```
Comment (frassinier) [inline src/Layout/Layout.scss]: Please, run sass-lint

Comment (jmfrancois) [inline src/Layout/Layout.scss]: no errors

File: src/Layout/TwoColumns/TwoColumns.scss [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+.container {
+	display: flex;
+	flex: 1 0 100%;
+	min-height: 0;
+}
+
+.sidemenu {
+	background-color: $tc-layout-two-columns-sidemenu-bg;
+
+	:global(.nav-pills) :global(.btn-link) {
```
Comment (jsomsanith) [inline src/Layout/TwoColumns/TwoColumns.scss]: This is already managed by SidePanel

File: src/Layout/TwoColumns/TwoColumns.scss [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+		color: #FFF;
+		width: 100%;
+	}
+
+	:global(.btn-link):hover {
```
Comment (jsomsanith) [inline src/Layout/TwoColumns/TwoColumns.scss]: This is already managed by SidePanel

**Pull request merged**